From 38b56458bc842e00cb4758f6b3d7dad6d8068af4 Mon Sep 17 00:00:00 2001
From: Han Xu <b45815@freescale.com>
Date: Fri, 9 Oct 2015 11:19:01 -0500
Subject: [PATCH 0720/5242] MLK-11720: mtd: gpmi: change the erase threshold
 for mx6qp,mx6ul and mx7d

commit  6241a07500ebffa254243d9be72441a5c07f2eec from
https://source.codeaurora.org/external/imx/linux-imx.git

The erase threshold should be set to ecc_strength for these platforms.

Signed-off-by: Han Xu <b45815@freescale.com>
(cherry picked from commit f46d113a02f5375c38fc9aba88c587fd672a30c4)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mtd/nand/raw/gpmi-nand/gpmi-lib.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/nand/raw/gpmi-nand/gpmi-lib.c b/drivers/mtd/nand/raw/gpmi-nand/gpmi-lib.c
index e95a185..97e2285 100644
--- a/drivers/mtd/nand/raw/gpmi-nand/gpmi-lib.c
+++ b/drivers/mtd/nand/raw/gpmi-nand/gpmi-lib.c
@@ -328,9 +328,9 @@ int bch_set_geometry(struct gpmi_nand_data *this)
 			| BF_BCH_FLASH0LAYOUT1_DATAN_SIZE(blockn_size, this),
 			r->bch_regs + HW_BCH_FLASH0LAYOUT1);
 
-	/* Set erase threshold to gf/2 for mx6ul, mx6qp and mx7 */
+	/* Set erase threshold to ecc strength for mx6ul, mx6qp and mx7 */
 	if (GPMI_IS_MX6QP(this) || GPMI_IS_MX7D(this) || GPMI_IS_MX6UL(this))
-		writel(BF_BCH_MODE_ERASE_THRESHOLD(gf_len/2),
+		writel(BF_BCH_MODE_ERASE_THRESHOLD(ecc_strength),
 			r->bch_regs + HW_BCH_MODE);
 
 	/* Set *all* chip selects to use layout 0. */
-- 
1.7.9.5

